src/parser/jobs/ast/modules/ArcanaTracking/ArcanaTracking.tsx

```diff
@@ -198,59 +177,55 @@ export default class ArcanaTracking extends Module {
+			this.on_prepullArcanas.status.push(event)
+		}
 	}
 
 	/**
 	 * Determine exactly when they casted their prepull arcana
 	 * If they had overwritten this buff, it will falsly pull back the timestamp of their prepull cast, but since we are guessing, it may as well be the same.
 	 */
-	private offPrepullArcana(event: BuffEvent) {
-		if (event.timestamp >= this.parser.fight.start_time + this.data.statuses.THE_BALANCE.duration) {
+	private offPrepullArcana(event: Events['statusRemove']) {
+		if (event.timestamp >= this.parser.pull.timestamp + this.data.statuses.THE_BALANCE.duration) {
 			return
 		}
+		if (this.off_prepullArcanas != null) {
+			this.off_prepullArcanas.status.forEach(arcanaBuff => {
+				if (!(arcanaBuff.status === event.status
+				&& arcanaBuff.target === event.target)) { return }
+
+				const cardStateItem: CardState = {..._.last(this.cardStateLog)} as CardState
+				const arcanaAction = getDataBy(this.data.actions, 'id', this.arcanaStatusToPlay(event.status))
+
+				if (arcanaAction == null) { return }
+
+				const arcanaCastEvent: Events['action'] = {
+					action: arcanaAction.id as number,
+					timestamp: event.timestamp - this.data.statuses.THE_BALANCE.duration - this.parser.pull.timestamp,
+					type: 'action',
+					source: event.source,
+					target: event.target,
+				}
+				cardStateItem.lastEvent = {...arcanaCastEvent}
+				cardStateItem.drawState = undefined
+				cardStateItem.sealState = CLEAN_SEAL_STATE
 
-		this.prepullArcanas.forEach(arcanaBuff => {
-			if (!(arcanaBuff.ability.guid === event.ability.guid
-				&& arcanaBuff.targetID === event.targetID)) { return }
-
-			const cardStateItem: CardState = {..._.last(this.cardStateLog)} as CardState
-			const arcanaAction = getDataBy(this.data.actions, 'id', this.arcanaStatusToPlay(event.ability.guid))
-
-			if (!arcanaAction) { return }
-
-			const arcanaCastEvent: CastEvent = {
-				ability: {
-					name: arcanaAction.name,
-					guid: arcanaAction.id as number,
-					type: 1,
-					abilityIcon: _.replace(_.replace(arcanaAction.icon, 'https://xivapi.com/i/', ''), '/', '-'),
-				},
-				timestamp: event.timestamp - this.data.statuses.THE_BALANCE.duration,
-				type: 'cast',
-				sourceIsFriendly: true,
-				targetIsFriendly: true,
-				sourceID: event.sourceID,
-				targetID: event.targetID,
-			}
-			cardStateItem.lastEvent = {...arcanaCastEvent}
-			cardStateItem.drawState = undefined
-			cardStateItem.sealState = CLEAN_SEAL_STATE
-
-			this.cardStateLog.unshift(cardStateItem)
-			this.pullIndex++
+				this.cardStateLog.unshift(cardStateItem)
+				this.pullIndex++
 
-		})
+			})
+		}
 
 	}
 
 	// Just saves a class var for the last drawn status buff event for reference, to help minor arcana plays
-	private onDrawnStatus(event: BuffEvent) {
-		if (!this.DRAWN_ARCANA.includes(event.ability.guid)) {
+	private onDrawnStatus(event: Events['statusApply']) {
+		if (typeof event.data !== 'undefined' && !this.drawnArcana.includes(event.data)) {
```

Comment:
This should be checking against event.status (to verify the ID maps to the list of valid arcana statuses.  event.data on a statusApply event is for the number of stacks, which won't be set for arcana events.

Reply:
updated